packages/ab-testing/src/index.ts

```diff
@@ -25,29 +38,83 @@ function getModuloValue(experiment: string, userId: number | string): number {
     return crc32.calculate(String(userId), crc32.calculate(experiment)) % 100;
 }
 
+function validateAllocationFields(fields: AllocationFields, userProfile: UserProfile): boolean {
+    let itemMatchesCriteria = true;
+    for (const key in fields) {
+        if (!fields[key].includes(userProfile[key])) {
+            itemMatchesCriteria = false;
+        }
+    }
+
+    return itemMatchesCriteria;
+}
+
+const isAllocationV2 = (allocation: any): allocation is AllocationV2 =>
+    allocation.range !== undefined || allocation.fields !== undefined;
+
+function validateUserWithinAllocationRange(userSegmentNum: number, range: AllocationV1) {
+    for (const allocation of range) {
+        if (allocation[0] <= userSegmentNum && userSegmentNum < allocation[1]) {
+            return true;
+        }
+    }
+
+    return false;
+}
+
+export function validateAllocation(
+    cohort: Cohort,
+    userProfile: UserProfile,
+    configName: string,
+    userId: number | string
+): boolean {
+    const userSegmentNum = getModuloValue(configName, userId);
+    if (cohort.allocation) {
+        if (isAllocationV2(cohort.allocation)) {
+            const range = cohort.allocation.range;
+            const fields = cohort.allocation.fields;
+            if (validateUserWithinAllocationRange(userSegmentNum, range) || !range) {
```

Comment:
This line expects users to be able to pass fields only, and not define range (effectively including 0-100). This mimics force include at this point, let me know if this is a valid use case or if we should require range to be defined before defining fields.